Service is killed when webpack watch is done

trash opened the pull request:
This fixes #27 by making sure the service is killed when the watch
task emits the "watch-close" event.
This fix only works with webpack >= 2.3.0 otherwise it's effectively a
noop because the event will never be emitted.

trash commented on test/integration/index.spec.js:
```diff
@@ -71,6 +82,18 @@ describe('[INTEGRATION] index', function () {
     });
   });
 
+  it('kills the service when the watch is done', function (done) {
```
This test successfully fails without the addition of the `watch-close` callback

johnnyreilly commented:
Great to see this! Do you know why the test is failing?

piotr-oles commented:
The test fails because of timeouts... It looks like travis is very busy now. I will increase timeouts in the another pr to make build less non-deterministic

trash committed:
Service is killed when webpack watch is done

This fixes #27 by making sure the service is killed when the watch
task emits the "watch-close" event.
This fix only works with webpack >= 2.3.0 otherwise it's effectively a
noop because the event will never be emitted.

piotr-oles commented:
Please, pull from the master branch - I have fixed problem with tests timeout

piotr-oles commented:
We should also increase webpack dependency version to 2.3.0 - lower versions are not fully compatible with this plugin - it will work but with the bug. And user still can install it and use with lower version but npm/yarn will show him a warning - it looks like proper behavior :)
Files to change: `package.json` and `README.md`.

johnnyreilly commented:
Sounds good!

trash commented:
Rebased against master, updated the webpack peer dependency version, and it looks like the CI is passing.

piotr-oles commented:
Thank you for your work, I will publish it as v0.2.7 :)

trash commented:
@piotr-oles Would it be possible to get 0.2.7 published today? I was hoping to get an issue relying on this change resolved before the weekend. Thanks so much if that's possible :)

piotr-oles commented:
Sorry, I am not be able to deploy today. I will do it tomorrow I hope ;)

piotr-oles commented:
@trash - Just released as v0.2.7 :)

johnnyreilly commented:
Great!

trash commented:
Fantastic. Thanks!
